src/parser/statement.js

```diff
@@ -228,9 +226,10 @@ pp.parseForStatement = function (node) {
   this.state.labels.push(loopLabel);
 
   let forAwait = false;
-  if (this.hasPlugin("asyncGenerators") && this.state.inAsync && this.isContextual("await")) {
-      forAwait = true;
-      this.next();
+  if (this.isContextual("await") && this.state.inAsync) {
```

Comment:
Maybe switch the order here like it was before, checking the bool value first is maybe also faster in case it is false?
Or did you have something in mind by changing the order.